chat/fe/src/services/NotificationService.ts

```diff
@@ -1,13 +1,29 @@
 let notificationsEnabled = false;
 
+function checkNotificationPromise() {
+  try {
+    Notification.requestPermission().then();
+  } catch (err) {
+    console.error(err);
+    return false;
+  }
+
+  return true;
+}
+
 export function initializeNotificationService() {
-  notificationsEnabled = localStorage.getItem("notifications") === "true";
+  notificationsEnabled = localStorage.getItem('notifications') === 'true';
 
-  return Notification.requestPermission()
-    .then()
-    .catch((err) => {
-      console.error(err);
-    });
+  if (!('Notification' in window)) {
+    return;
```

Comment:
Safari IOS does not support Notification.requestPermission() so I added this early return;